docs/external-cloud-provider.md

```diff
@@ -0,0 +1,60 @@
+# External Cloud Provider
+
+To deploy a cluster using [external cloud provider](https://github.com/kubernetes-sigs/cloud-provider-openstack), create a cluster configuration with the [external cloud provider template](../../templates/cluster-template-external-cloud-provider.yaml).
+
+## Steps
+
+- After control plane is up and running, retrieve the workload cluster Kubeconfig:
+
+    ```shell
+    kubectl --namespace=default get secret/${CLUSTER_NAME}-kubeconfig -o jsonpath={.data.value} \
+    | base64 --decode \
+    > ./${CLUSTER_NAME}.kubeconfig
+    ```
+
+- Deploy a CNI solution
+
+    ```shell
+    curl https://docs.projectcalico.org/v3.12/manifests/calico.yaml | sed "s/veth_mtu:.*/veth_mtu: \"1430\"/g" | kubectl --kubeconfig=./${CLUSTER_NAME}.kubeconfig apply -f -
+    ```
+
+- Create a secret containing the cloud configuration
```

Comment:
maybe you want to share some info about why we need perform following .e.g
 the node is Ready but not schedulable due to taint
`node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule`

Reply:
I am sorry can not understand you want to point out.
I followed [cloud-provider-openstack doc](Deploy a Kubernetes cluster with openstack-cloud-controller-manager using kubeadm), I did not find `node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule` .

Reply:
yes, I knew, what I mean is at this stage ,if you describe the node
the node is `Ready` state, but pods are pending due to this `taint`

instead of let people go to other doc, we might notify them that at this stage what happend and why the pods is pending etc (so people who as developer might know more background )